### DISCUSSION TITLE
The ossl_rsa_sp800_56b_generate_key() cannot generate a 2049-bit key.

### DISCUSSION THREAD

**Discussion by jchx-c** [openssl/openssl#25577] [Q&A] [answered]
During RSA key generation, we expect to obtain a 2049-bit key but obtain n-bits = 2048 when call ossl_rsa_sp800_56b_generate_key(),  is this because the FIPS standard does not apply when n is an odd number?

**Comment by t8m** [comment, marked as answer]
The RSA key generation as specified in FIPS 186-5 (and older versions of the standard) cannot generate keys with odd number of bits.
